src/parser/Parser.ts

```diff
@@ -911,7 +911,7 @@ export class Parser {
                 }
                 thenBranch = new Stmt.Block([thenStatement], peek().location);
 
-                while (match(Lexeme.ElseIf)) {
+                while (previous().kind !== Lexeme.Newline && match(Lexeme.ElseIf)) {
```

Comment:
There's already a `checkPrevious()` method that does this in a slightly less verbose way. @sjbarag might ask you to use that.

Reply:
will take a look, thanks for the review